src/main/java/com/maxmind/minfraud/request/Email.java

```diff
@@ -14,6 +20,29 @@
     private final String address;
     private final boolean hashAddress;
     private final String domain;
+    private static final Map<String, String> typoDomains;
+    private static final Pattern addressDashRegex;
+    private static final Pattern addressPlusRegex;
+
+    static {
+        HashMap<String, String> m = new HashMap<>();
+
+        // gmail.com
+        m.put("35gmai.com", "gmail.com");
+        m.put("636gmail.com", "gmail.com");
+        m.put("gamil.com", "gmail.com");
+        m.put("gmail.comu", "gmail.com");
+        m.put("gmial.com", "gmail.com");
+        m.put("gmil.com", "gmail.com");
+        m.put("yahoogmail.com", "gmail.com");
+        // outlook.com
+        m.put("putlook.com", "outlook.com");
```

Comment:
This could be written as:

```
        new HashMap<String, String>() {{
            // gmail.com
            put("35gmai.com", "gmail.com");
            put("636gmail.com", "gmail.com");
            put("gamil.com", "gmail.com");
            put("gmail.comu", "gmail.com");
            put("gmial.com", "gmail.com");
            put("gmil.com", "gmail.com");
            put("yahoogmail.com", "gmail.com");
            // outlook.com
            put("putlook.com", "outlook.com");
        }};
```

I think it might be a bit cleaner as it groups everything better, but I don't have a strong opinion.

Reply:
Yeah, looks better!